asterisk/agi/library/Agi/Action/ExternalRetailCallAction.php

```diff
@@ -53,21 +53,15 @@ public function process()
                 $this->agi->setRedirecting('from-num', $ddi->getDDIE164());
             }
         } else {
-            // Allow identification from any company DDI
+            // Allow identification from any Retail Account DDI
             $callerIdNum = $retail->preferredToE164($this->agi->getCallerIdNum());
-            $retailDDIs = $retail->getDDIs();
-            foreach ($retailDDIs as $retailDDI) {
-                if ($callerIdNum === $retailDDI->getDDIE164()) {
-                    $this->agi->notice("Retail account \e[0;36m%s [retail%d]\e[0;93m presented origin matches company DDI %s [ddi%d].",
-                            $retail->getName(), $retail->getId(), $retailDDI->getDDIE164(), $retailDDI->getId());
-                    $ddi = $retailDDI;
-                    // FIXME, please kamuser, give me the numbers in E.164
-                    $this->agi->setCallerIdNum($ddi->getDDIE164());
-                    break;
-                }
+            $ddi = $retail->getDDI($callerIdNum);
+            if (!empty($ddi)) {
+                $this->agi->notice("Retail account \e[0;36m%s [retail%d]\e[0;93m presented origin matches account DDI %s [ddi%d].",
+                        $retail->getName(), $retail->getId(), $ddi->getDDIE164(), $ddi->getId());
```

Comment:
Mmm.. not sure what you mean here... something like this?
```
        $this->agi->notice("Retail account \e[0;36m%s [retail%d]\e[0;93m presented origin matches account DDI %s [ddi%d].",
            $retail->getName(), $retail->getId(), $ddi->getDDIE164(), $ddi->getId());
```

or something like this:

```
        $this->agi->notice("Retail account \e[0;36m%s [retail%d]\e[0;93m presented origin matches account DDI %s [ddi%d].",
            $retail->getName(),
            $retail->getId(),
            $ddi->getDDIE164(),
            $ddi->getId());
```